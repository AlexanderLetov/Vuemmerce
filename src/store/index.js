import Vue from 'vue';
import Vuex from 'vuex';
import initalState from '../initial-state.json';
import search from './modules/search';

Vue.use(Vuex);

export default new Vuex.Store({
  strict: true,

  state: initalState,

  getters: {
    getCategoryById: state => id => {
      return state.categories.find(category => category.id == id);
    },
    getProductsList: state => {
      return state.products;
    },
    productsAdded: state => {
      return state.products.filter(el => {
        return el.isAddedToCart;
      });
    },
    productsAddedToFavourite: state => {
      return state.products.filter(el => {
        return el.isFavourite;
      });
    },
    getProductById: state => id => {
      return state.products.find(product => product.id == id);
    },
    getProductImages: state => id => {
      return state.products.find(product => product.id == id).images;
    },
    isUserLoggedIn: state => {
      return state.userInfo.isLoggedIn;
    },
    isUserSignedUp: state => {
      return state.userInfo.isSignedUp;
    },
    getUserName: state => {
      return state.userInfo.name;
    },
    getUserEmail: state => {
      return state.userInfo.email;
    },
    getUserAddress: state => {
      return state.userInfo.address;
    },
    getUserPassword: state => {
      return state.userInfo.password;
    },
    getAvatar: state => {
      return state.userInfo.avatarBinary;
    },
    isLoginModalOpen: state => {
      return state.systemInfo.openLoginModal;
    },
    isSignupModalOpen: state => {
      return state.systemInfo.openSignupModal;
    },
    isCheckoutModalOpen: state => {
      return state.systemInfo.openCheckoutModal;
    },
    quantity: state => {
      return state.products.quantity;
    },
    getNewsById: state => id => {
      return state.news.find(newItem => newItem.id == id);
    }
  },

  actions: {
    addToCartAction ({ commit }, id) {
      return new Promise((resolve, reject) => {
          commit('addToCart', id);
          commit('setAddedBtn', id);
          resolve();
      })
    }
  },

  mutations: {
    addToCart: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isAddedToCart = true;
        }
      });
    },
    setAddedBtn: (state, data) => {
      state.products.forEach(el => {
        if (data.id === el.id) {
          el.isAddedBtn = data.status;
        }
      });
    },
    removeFromCart: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isAddedToCart = false;
        }
      });
    },
    removeProductsFromFavourite: state => {
      state.products.filter(el => {
        el.isFavourite = false;
      });
    },
    isUserLoggedIn: (state, isUserLoggedIn) => {
      state.userInfo.isLoggedIn = isUserLoggedIn;
    },
    isUserSignedUp: (state, isSignedUp) => {
      state.userInfo.isSignedUp = isSignedUp;
    },
    setUserName: (state, name) => {
      state.userInfo.name = name;
    },
    saveAvatar: (state, avatarBinary) => {
      state.userInfo.avatarBinary = avatarBinary;
    },
    setHasUserSearched: (state, hasSearched) => {
      state.userInfo.hasSearched = hasSearched;
    },
    setUserName: (state, name) => {
      state.userInfo.name = name;
    },
    setUserEmail: (state, email) => {
      state.userInfo.email = email;
    },
    setUserAddress: (state, address) => {
      state.userInfo.address = address;
    },
    setUserPassword: (state, password) => {
      state.userInfo.password = password;
    },
    setProductTitleSearched: (state, titleSearched) => {
      state.userInfo.productTitleSearched = titleSearched;
    },
    showLoginModal: (state, show) => {
      state.systemInfo.openLoginModal = show;
    },
    showSignupModal: (state, show) => {
      state.systemInfo.openSignupModal = show;
    },
    showCheckoutModal: (state, show) => {
      state.systemInfo.openCheckoutModal = show;
    },
    addToFavourite: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isFavourite = true;
        }
      });
    },
    removeFromFavourite: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isFavourite = false;
        }
      });
    },
    quantity: (state, data) => {
      state.products.forEach(el => {
        if (data.id === el.id) {
          el.quantity = data.quantity;
        }
      });
    }
  },

  modules: {
    search
  }
})
