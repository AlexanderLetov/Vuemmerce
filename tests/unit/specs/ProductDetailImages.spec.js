import { shallowMount, mount } from "@vue/test-utils";
import { createRenderer } from 'vue-server-renderer'
import component from "../../../src/components/product_detail/ProductDetailImages.vue";
import store from "../../../src/store";
import router from "../../../src/router";

const localVue = global.newVueInstance();

describe('ProductDetailImages.vue', async () => {
  it('start mount', () => {
    const wrapper = mount(component, {
        localVue,
        router,
        store,
        propsData: { slides: [], productId: 1 }
    });

    let element = wrapper.find('.hooper');
    expect(element.is('section')).toBe(true);
  });

   it('loading images', () => {
     const wrapper = mount(component, {
         localVue,
         router,
         store,
         propsData: { slides: [], productId: 1 }
         });
     expect(wrapper.vm.slides.length).toBeGreaterThan(0);
   });

   it('snapshot renderer test', () => {
    const renderer = createRenderer()
    const wrapper = mount(component, {
       localVue,
       router,
       store,
       propsData: { slides: [], productId: 1 }
    });
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    });
  });
});
