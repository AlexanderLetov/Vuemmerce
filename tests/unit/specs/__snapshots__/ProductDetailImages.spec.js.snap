// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailImages.vue snapshot renderer test 1`] = `
<div slides="">
  <section tabindex="0" class="carousel hooper">
    <div class="hooper-list">
      <ul class="hooper-track" style="transform:translate(0px, 0);">
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #1
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #2
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #3
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #4
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #5
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #6
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #7
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #8
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #9
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-prev" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #10
          </div>
        </li>
        <li class="hooper-slide is-active" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #1
          </div>
        </li>
        <li class="hooper-slide is-active" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #2
          </div>
        </li>
        <li class="hooper-slide is-active" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #3
          </div>
        </li>
        <li class="hooper-slide is-active" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #4
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-next" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #5
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #6
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #7
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #8
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #9
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #10
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #1
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #2
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #3
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #4
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #5
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #6
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #7
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #8
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #9
          </div>
        </li>
        <li aria-hidden="true" class="hooper-slide is-clone" style="width:0px;">
          <div style="height: 128px; width: 100%; text-align: center; background-color: #FAFAFA;
                    background-image: url(https://bulma.io/images/placeholders/128x128.png); background-repeat: no-repeat; background-position: center center;">
            Slide #10
          </div>
        </li>
      </ul>
      <div class="hooper-navigation"><button type="button" class="hooper-prev"><svg class="icon icon-arrowLeft" viewBox="0 0 24 24" width="24px" height="24px">
            <title>Arrow Left</title>
            <path d="M0 0h24v24H0z" fill="none"></path>
            <path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"></path>
          </svg></button><button type="button" class="hooper-next"><svg class="icon icon-arrowRight" viewBox="0 0 24 24" width="24px" height="24px">
            <title>Arrow Right</title>
            <path d="M0 0h24v24H0z" fill="none"></path>
            <path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"></path>
          </svg></button></div>
      <div aria-live="polite" aria-atomic="true" class="hooper-liveregion hooper-sr-only">Item 1 of 10</div>
    </div>
  </section>
</div>
`;
